### PR TITLE
[libigl] update to 2.4.0

### LINKED ISSUE #23354: [libigl] update to 2.4.0
Library name: `libigl`

New version number: 2.4.0

There are several **breaking changes**: https://github.com/libigl/libigl/releases/tag/v2.4.0
It might need additional modification of the port file, which I'm not capable of.
Would it be possible to update it?

Thank you.

### PR DESCRIPTION
**Pull request by FrankXie05**
Fix #23354

Update [libigl] version to 2.4.0

### CODE REVIEW

**Comment by JonLiu1993** [comment]
@FrankXie05, Is this PR being worked on?

**Comment by FrankXie05** [comment]
@JonLiu1993 Yes, I will fix this PR completely next week.  :)

**Comment by JonLiu1993** [comment]
Pinging @FrankXie05  for response. Is work still being done for this PR?

File: ports/libigl/use-hunter-eigen.patch [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+-    target_sources(Eigen3_Eigen INTERFACE $<BUILD_INTERFACE:${eigen_SOURCE_DIR}/debug/msvc/eigen.natvis>)
+-endif()
+-
+-# Install rules
+-set(CMAKE_INSTALL_DEFAULT_COMPONENT_NAME eigen)
+-set_target_properties(Eigen3_Eigen PROPERTIES EXPORT_NAME Eigen)
+-install(DIRECTORY ${eigen_SOURCE_DIR}/Eigen DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+-install(TARGETS Eigen3_Eigen EXPORT EigenTargets)
+-install(EXPORT EigenTargets DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/eigen NAMESPACE Eigen3::)
++find_package(Eigen3 CONFIG REQUIRED)
```
Comment (dg0yt) [inline ports/libigl/use-hunter-eigen.patch:49]: Why not do this early and `return()`, leaving the rest of the file unchanged?

**Review by dg0yt — CHANGES_REQUESTED** [review]
> use hunter eigen

Why not vcpkg eigen?

File: ports/libigl/use-hunter-eigen.patch (hunk 1 above)
Comment (FrankXie05) [reply ports/libigl/use-hunter-eigen.patch:49]: We need to enable the hunter function to use the package provided by vcpkg. I'm doing this. https://github.com/libigl/libigl/tree/main/cmake/recipes/hunter

Comment (dg0yt) [reply ports/libigl/use-hunter-eigen.patch:49]: Hm, I think you missed the point. I suggested to minimize that patch.

**Comment by iAnyKey** [comment]
for some reason it wont compile on windows

**Comment by iAnyKey** [comment]
> for some reason it wont compile on windows

fixed with https://github.com/FrankXie05/vcpkg/pull/4

**Comment by FrankXie05** [comment]
Duplicate of https://github.com/microsoft/vcpkg/pull/27963
Let's move to this [PR ](https://github.com/microsoft/vcpkg/pull/27963) to fix.

**Pull request closed without merging**